[Feat] 이슈 확인 기능을 구현한다.

asd42270 opened the pull request:
# Pull Request 요약

- 이슈 확인 기능을 구현한다.

## 변경 사항

- 이슈 확인 기능 코드 작성했습니다.
- 테스트 코드 및 아스키닥 문서도 작성 완료했습니다!!
- 상세 페이지로 넘어가기 전 이슈가 있을 경우 요청되는 api입니다. 
- 해당 api를 통해 issue 내용을 전달합니다!

## 관련 이슈

- #106 

## 개발 유형

- [ ] 버그 수정
- [x] 새로운 기능 개발
- [ ] 리팩토링
- [x] 테스트 코드 작성
- [x] 문서 업데이트
- 기타 (아래에 설명 추가)

## 작업 기간

- 작업 시작일: (2024-05-28)
- 작업 종료일: (2024-05-28)

## 유의사항

- 나중에 리팩토링할 때 만약 알림 기능에 읽지 않은 이슈 여부까지 알림으로 전달해주게 되면 이슈 확인 save()를 이슈를 발행하면서 같이 save()하고 이슈를 읽었을 때 읽은 정보를 update해주는 방법으로 진행하는 것까지 생각해봤습니다!
- ManageArticle manageArticle = manageArticleRepository.findById(manageArticleId)
                .orElseThrow(ArticleNotFoundException::new); 이 부분은 우선 db에서 조회하는 방식으로 처리했습니다. 추후에 캐시를 사용하게 된다면 캐시 담아놨다가 캐시에서 가져오는 방법으로 리팩토링하면 성능이 더 좋아질 것 같습니다!!

asd42270 committed:
[feat] 이슈 확인 기능을 구현한다.
- ArticleIssueCheck: 컬럼이름 checked로 변경
- ArticleIssueCheckRepository: repository 생성
- ErrorCode: ISSUE_NOT_FOUND 에러 추가
- IssueNotFoundException: 에러 코드 추가

asd42270 committed:
[feat] 이슈 확인 기능을 구현한다.
- ArticleIssueCheckRequestDto: @AssertTrue로 검증 적용
- ArticleIssueCheckController: 컨트롤러 클래스 구현

asd42270 committed:
[feat] 이슈 확인 기능의 테스트 코드를 작성한다.
- IssueCheckServiceTest: 성공, 실패 케이스에 대한 테스트 코드를 각각 작성한다.

asd42270 committed:
[feat] 이슈 확인 기능의 컨트롤러 테스트 코드를 작성한다.
- ArticleIssueCheckController: hasRole의 OWNER-> MEMBER로 변경한다.
- ArticleIssueCheckControllerDocsTest: 아스키닥 작성을 위한 테스트 코드를 작성한다.

asd42270 committed:
[feat] 이슈 확인 기능의 아스키닥 문서를 작성한다.

asd42270 committed:
[fix] request 패키지 변경으로 인한 경로 수정

asd42270 committed:
[fix] ArticleIssueCheckRequestDto 저장 위치 변경으로 인한 경로 수정
- ArticleIssueCheck: 이슈가 발행된 직후 아무도 읽지 않은 상태이므로 default로 false 지정
- 그 외: requestDto의 경로 변경

Galmaeki commented on src/main/java/com/final_10aeat/domain/articleIssue/dto/request/ArticleIssueCheckRequestDto.java:
```diff
@@ -0,0 +1,9 @@
+import jakarta.validation.constraints.AssertTrue;
+
+public record ArticleIssueCheckRequestDto(
+        @AssertTrue
```
어떤 용도로 사용하신 어노테이션인지 여쭤봐도 될까요?

asd42270 replied:
확인했어요 버튼을 눌렀을 때 무조건 true 값만 반환하도록 하기 위해서 해당 어노테이션을 사용했습니다! 혹시 용도가 잘못됐다면 바꾸겠습니다!!

Galmaeki replied:
일정한 값만 받길 원한다면 벨리데이션 보다는 바디를 없애도 좋을것 같았어요!
이건 다른 값은 없으니까 바디가 없어도 좋을것 같다고 생각합니다!

asd42270 replied:
음 좋은 것 같습니담 안 그래도 request값으로 하나만 받는데 굳이 dto로 감싸야 하나 싶기는 했는데..ㅋㅋㅋㅋ 알겠습니다!!

Galmaeki reviewed (changes requested):
고생하셨습니다!
코멘트 확인 부탁드림다!

asd42270 committed:
[feat] 이슈 확인 기능을 구현한다.
- ArticleIssueCheck: checked의 deafult를 삭제한다.
- ArticleIssueCheckController: 응닶 값 생성으로 return값을 변경하고, url을 세부적으로 나누어 구분한다.
- ArticleIssueCheckResponseDto: 응답 dto를 생성한다.
- ArticleIssueService: 서비스 클래스 생성
- ArticleIssueRepository: 가장 최근에 등록된 이슈를 조회하는 메서드 추가

ghrltjdtprbs commented on src/docs/asciidoc/articleissue/issuecheck-api.adoc:
```diff
@@ -0,0 +1,24 @@
+= Repair Article REST API Docs
```
제목 변경 누락된 것 같습니다!

asd42270 replied:
넵 수정했습니다! 감사합니다

ghrltjdtprbs commented on src/test/java/com/final_10aeat/domain/issue/docs/ArticleIssueCheckControllerDocsTest.java:
구현한 이슈 api는 두개인데, 왜 테스트 코드를 한 개만 구현하셨는지 궁금합니다!

asd42270 replied:
아 제가 커밋해놓고 푸시를 안 했었던 것 같습니다 ㅋㅋㅋ...지금 다시 커밋하고 푸시까지 했습니다!!

ghrltjdtprbs replied:
ㅋㅋㅋㅋㅋ굿굿👍👍

ghrltjdtprbs reviewed (approved):
리뷰 확인 부탁드립니다! 혹시 전부 postman으로 작동 확인 하셨나요???

asd42270 committed:
[Refactor] 이슈 확인 기능의 누란된 테스트 코드 커밋을 재커밋한다.
- ArticleIssueCheckControllerDocsTest: 테스트 코드 재커밋
- issuecheck-api.adoc: api문서 재커밋
- ManageIssueCheckServiceTest, RepairIssueCheckServiceTest: 테스트코드 구분하여 작성

asd42270 committed:
[Refactor] 이슈 확인 기능의 누란된 테스트 코드 커밋을 재커밋한다.
- ArticleIssueCheckControllerDocsTest: 중복 코드 삭제
- RepairIssueCheckServiceTest: import문 수정

asd42270 committed:
[Refactor] api문서의 제목을 수정한다.
- issuecheck-api.adoc: 제목 수정

asd42270 commented:
> 리뷰 확인 부탁드립니다! 혹시 전부 postman으로 작동 확인 하셨나요???

이슈 기능은 아직 안 해봤습니다...ㅋㅋ...해 보고 안 되는 부분 수정해서 머지할게요!!

asd42270 committed:
[Refactor] 권한 어노테이션을 수정한다.
- MEMBER -> USER로 수정

asd42270 committed:
Merge branch 'develop' into feature/106
